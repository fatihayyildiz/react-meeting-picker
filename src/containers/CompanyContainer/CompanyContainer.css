.top-sticky {
	position: -webkit-sticky;
	position: sticky;
	top: 0;
	background-color: white;
}
