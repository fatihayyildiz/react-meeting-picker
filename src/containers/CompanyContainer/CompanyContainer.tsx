import React, { useEffect, useState } from 'react';
import DaySlots, { DaySlotsProps } from 'containers/DaySlots/DaySlots';
import ReservationCard from 'components/ReservationCard/ReservationCard';
import { useSelector } from 'react-redux';
import { slotSelector } from 'store/slots/slice';
import 'containers/CompanyContainer/CompanyContainer.css';
import { CompanySelectedSlot } from '../../store/slots/types';

export type CompanyContainerProps = {
	id: number;
	name: string;
	days: Array<DaySlotsProps>;
};

const CompanyContainer = ({ id, name, days }: CompanyContainerProps) => {
	const [selectedSlot, setSelectedSlot] = useState<CompanySelectedSlot>();

	const { selectedCompanySlots } = useSelector(slotSelector);

	useEffect(() => {
		if (selectedCompanySlots.length > 0) {
			const foundCompanyIndex = selectedCompanySlots.findIndex(
				(slot) => slot.id === id
			);
			if (foundCompanyIndex > -1) {
				console.log('Selected company slot:', selectedCompanySlots);
				setSelectedSlot(selectedCompanySlots[foundCompanyIndex]);
			} else setSelectedSlot(undefined);
		} else setSelectedSlot(undefined);
	}, [selectedCompanySlots, id]);

	return (
		<div className="container">
			<div className="top-sticky">
				<div className="row centered-box">
					<span>{name}</span>
				</div>
				{selectedSlot && (
					<div className="row centered-box">
						<ReservationCard
							id={id}
							day={selectedSlot.selectedSlot.day}
							start_time={selectedSlot.selectedSlot.start_time}
							end_time={selectedSlot.selectedSlot.end_time}
						/>
					</div>
				)}
			</div>
			<div className="row">
				{days.map((daySlots, index) => {
					return (
						<DaySlots
							key={`day-${index}`}
							id={id}
							day={daySlots.day}
							slots={daySlots.slots}
						/>
					);
				})}
			</div>
		</div>
	);
};

export default CompanyContainer;
