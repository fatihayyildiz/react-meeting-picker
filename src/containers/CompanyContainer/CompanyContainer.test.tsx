import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CompanyContainer from 'containers/CompanyContainer/CompanyContainer';
import SingleFormattedCompany from 'mocks/resources/single_formatted_company.json';

afterEach(cleanup);

const testProps = {
	name: SingleFormattedCompany[0].name,
	days: SingleFormattedCompany[0].days,
};

test('Company Container should match with snapshot', () => {
	// First render component and get as fragment
	const { asFragment } = render(
		<CompanyContainer {...{ name: testProps.name, days: testProps.days }} />
	);

	// Compare with saved snapshot
	expect(asFragment()).toMatchSnapshot();
});
