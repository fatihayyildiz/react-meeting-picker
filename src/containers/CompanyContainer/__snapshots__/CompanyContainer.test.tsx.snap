// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Container should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="top-sticky"
    >
      <div
        class="row centered-box"
      >
        <span>
          company1
        </span>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="container centered-box"
      >
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            18.03 - Monday
          </div>
        </div>
        <div
          class="row g-0 centered-box no-border"
        >
          <div
            class="row time-slot-box"
            data-testid="time-slot"
          >
            <div
              class="col-4"
            >
              08:00
            </div>
            <div
              class="col-2"
            >
              -
            </div>
            <div
              class="col-4"
            >
              09:00
            </div>
          </div>
          <div
            class="row time-slot-box"
            data-testid="time-slot"
          >
            <div
              class="col-4"
            >
              10:00
            </div>
            <div
              class="col-2"
            >
              -
            </div>
            <div
              class="col-4"
            >
              11:00
            </div>
          </div>
        </div>
      </div>
      <div
        class="container centered-box"
      >
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            19.03 - Tuesday
          </div>
        </div>
        <div
          class="row g-0 centered-box no-border"
        >
          <div
            class="row time-slot-box"
            data-testid="time-slot"
          >
            <div
              class="col-4"
            >
              12:00
            </div>
            <div
              class="col-2"
            >
              -
            </div>
            <div
              class="col-4"
            >
              13:00
            </div>
          </div>
          <div
            class="row time-slot-box"
            data-testid="time-slot"
          >
            <div
              class="col-4"
            >
              14:00
            </div>
            <div
              class="col-2"
            >
              -
            </div>
            <div
              class="col-4"
            >
              16:00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
